File: backend/src/main/java/ru/sportexpress/menuApp/service/MenuService.java
package ru.sportexpress.menuApp.service;

import ru.sportexpress.menuApp.entity.Menu;

import java.util.List;
import java.util.Optional;

public interface MenuService {
    List<Menu> getAll();

    Optional<Menu> getOne(Integer id);

    Menu update(int id, Menu customer);

    Menu save(Menu customer);

    Menu delete(int id);
}

File: backend/src/main/kotlin/ru/sportexpress/menuApp/web/service/MenuService.kt
package ru.sportexpress.menuApp.web.service

import org.springframework.http.ResponseEntity
import ru.sportexpress.menuApp.entity.Menu
import ru.sportexpress.menuApp.requests.MenuResponse
import java.util.*


public interface MenuService {
    fun getAll(): MenuResponse

    fun getOne(id : Long): Optional<Menu>

    fun update(id : Long, customer : Menu):Menu

    fun save(customer : Menu):Menu

    fun delete(id : Long): ResponseEntity<*>
}
